src/rendergraph/common/rendergraph/attributeset.h

```diff
@@ -4,17 +4,17 @@
 #include "rendergraph/attributeinit.h"
 
 namespace rendergraph {
-class AttributeSet;
-}
 
-class rendergraph::AttributeSet : public rendergraph::BaseAttributeSet {
+class AttributeSet : public BaseAttributeSet {
   public:
     AttributeSet(std::initializer_list<AttributeInit> list, const std::vector<QString>& names);
 };
 
-namespace rendergraph {
-template<typename... T>
-AttributeSet makeAttributeSet(const std::vector<QString>& names) {
-    return AttributeSet({(AttributeInit::create<T>())...}, names);
+template<typename... Ts, int N>
+AttributeSet makeAttributeSet(const QString (&names)[N]) {
+    static_assert(sizeof...(Ts) == N);
```

Comment:
```suggestion
AttributeSet makeAttributeSet(const QString (&names)[N]) {
    static_assert(sizeof...(Ts) == N, "you can add a user-friendly error message here");
```
This probably makes sense because the actual expression is not easy to understand.